src/Data/Time/PreciseDuration.purs

```diff
@@ -12,58 +14,85 @@ module Data.Time.PreciseDuration
   , toWeeks
   ) where
 
-import Prelude (($), (/), (<<<), (<>))
+import Prelude (($), (/), (>>>), (<<<), (<>))
 
+import Data.BigInt (BigInt)
 import Data.BigInt as BigInt
 import Data.Decimal (Decimal)
 import Data.Decimal as Decimal
 
-import Data.Time.PreciseDuration.Internal (PreciseDuration(..), bigIntToDecimal, day, hour, micro, milli, minute, second, toNanosecondsBI, toNanosecondsD, week)
-import Data.Time.PreciseDuration.Internal (PreciseDuration(..)) as PreciseDurationType
+import Data.Time.PreciseDuration.Internal (PreciseDuration(..), day, hour, micro, milli, minute, second, toNanosecondsD, week)
+import Data.Time.PreciseDuration.Internal (PreciseDuration) as PreciseDurationType
 
 toString :: PreciseDuration -> String
-toString =
-  case _ of
-    Nanoseconds d -> BigInt.toString d <> "ns"
-    Microseconds d -> Decimal.toString d <> "us"
-    Milliseconds d -> Decimal.toString d <> "ms"
-    Seconds d -> Decimal.toString d <> "s"
-    Minutes d -> Decimal.toString d <> "m"
-    Hours d -> Decimal.toString d <> "h"
-    Days d -> Decimal.toString d <> "d"
-    Weeks d -> Decimal.toString d <> "w"
+toString = case _ of
+  Nanoseconds d -> Decimal.toString d <> "ns"
+  Microseconds d -> Decimal.toString d <> "us"
+  Milliseconds d -> Decimal.toString d <> "ms"
+  Seconds d -> Decimal.toString d <> "s"
+  Minutes d -> Decimal.toString d <> "m"
+  Hours d -> Decimal.toString d <> "h"
+  Days d -> Decimal.toString d <> "d"
+  Weeks d -> Decimal.toString d <> "w"
+
+-- Smart constructors
+make = { nanoseconds, microseconds, milliseconds, seconds, minutes, hours, days, weeks }
+
+unsafeNanoseconds :: Decimal -> PreciseDuration
+unsafeNanoseconds = Nanoseconds
+
+nanoseconds :: Int -> PreciseDuration
+nanoseconds = Decimal.fromInt >>> Nanoseconds
+
+microseconds :: Decimal -> PreciseDuration
+microseconds = Microseconds
+
+milliseconds :: Decimal -> PreciseDuration
+milliseconds = Milliseconds
+
+seconds :: Decimal -> PreciseDuration
+seconds = Seconds
+
+minutes :: Decimal -> PreciseDuration
+minutes = Minutes
+
+hours :: Decimal -> PreciseDuration
+hours = Hours
+
+days :: Decimal -> PreciseDuration
+days = Days
+
+weeks :: Decimal -> PreciseDuration
+weeks = Weeks
 
 unPreciseDuration :: PreciseDuration -> Decimal
-unPreciseDuration = case _ of
-  Nanoseconds d -> bigIntToDecimal d
-  Microseconds d -> d
-  Milliseconds d -> d
-  Seconds d -> d
-  Minutes d -> d
-  Hours d -> d
-  Days d -> d
-  Weeks d -> d
+unPreciseDuration = toNanosecondsD
```

Comment:
Not sure why this function exists _and_ `toNanoseconds` exists.

Reply:
```
toNanoseconds :: PreciseDuration -> PreciseDuration
unPreciseDuration :: PreciseDuration -> Decimal
```